File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        result = []

        nums.sort()
        n = len(nums)
        i = 0

        while i < n:
            j = i + 1
            k = n - 1

            while j < n and k > 0 and j < k:
                sum = nums[i] + nums[j] + nums[k]

                if sum == 0:
                    result.append([nums[i], nums[j], nums[k]])
                    j = j + 1
                    k = n - 1
                    while j < n and nums[j] == nums[j-1]:
                        j = j + 1

                else:
                    if sum < 0:
                        j = j + 1
                    else:
                        k = k - 1
            i = i + 1            
            while i < n and nums[i] == nums[i-1]:
                i = i + 1 

        return result

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
    static bool cmp(pair<int, int>& a, pair<int, int>& b) {
        return a.second > b.second;
    }

    static vector<pair<int, int>> sortmap(map<int, int>& M) {
        vector<pair<int, int>> A;

        for (auto& it : M) {
            A.push_back(it);
        }

        sort(A.begin(), A.end(), cmp);
        return A;
    }
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> result;
        map<int, int> m;
        for (int i = 0; i < nums.size(); i++) {
            if (m.count(nums[i])) {
                m[nums[i]]++;
            } else {
                m[nums[i]] = 0;
            }
        }

        vector<pair<int, int>> a = sortmap(m);

        for (auto const& x : a) {
            if (k == 0) {
                return result;
            }
            result.push_back(x.first);
            k--;
        }

        return result;
    }
};
